components/annot-discrete-choices.vue

```diff
@@ -236,14 +240,22 @@
             },
 
             updateMapping(p_selectedValue, p_row) {
-
-                // 1. Update the local annotation value map with the selected, new annotation value
-                this.valueMapping[p_row.column_name][p_row.raw_value] = p_selectedValue;
-
-                // 2. Determine whether all unique values have now been mapped to something
-                // and set the disabled status of the save annotation button
-                // TODO: This might be better suited for a computed property, but this seems to break reactivity
-                this.saveButtonDisabled = !this.checkAnnotationState();
+                // This method updates the valueMapping object.
+                // In order for Vue to detect this change and be reactive, we have to merge the new
+                // value with the existing object, one level at a time
+
+                // First, we merge the inner level (e.g. the mapping for the columnName)
+                const innerUpdate = Object.assign(
```

Comment:
sorry, what's the first part?

Reply:
It begins "Can you explain the difference between". There are two questions in the comment.

Reply:
Ah I see. Check [out Object.assign()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign). You can also try it out directly (e.g. [here](https://replit.com/languages/nodejs)). The bottom line is that this code handles new value mappings one at a time reactively. That is, we need to fulfil two conditions 
every time a new value is mapped:
1. add the new mapped value to the global store object keyed under `store.object[columnName][newValue]`. AND
2. make sure any previously mapped values are still there and not lost / overwritten

Your second example fulfils only 1, but not 2 because `Object.assign` does not do a deep merge and you would overwrite the existing entried under `store.object[columName]` with your new single key object.

I think we could write our own recursive deep merge method if we keep having to do this ( I think we're already using it twice in the codebase).

Reply:
Yes, you're referring to `structuredClone`. That's very particular behavior. It appears only shared keys from the source argument objects are assigned to the target.

Reply:
Feel free to see if we can do it with `structuredClone` in a single command. I [went off of what the docs suggested in order to ensure reactivity of the assignment](https://v2.vuejs.org/v2/guide/reactivity.html#For-Objects). If `structuredClone` is more intuitive or simpler, by all means let's use it!